packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -537,6 +542,10 @@ export class Cluster extends Resource implements ICluster {
       machineImageType: options.machineImageType,
     });
 
+    if (!this._neuronDevicePlugin && nodeTypeForInstanceType(options.instanceType) === NodeType.INFERENTIA) {
```

Comment:
Am I missing something? This is the constructor. By definition `_neuronDevicePlugin` would be `undefined` at this point, no?

```suggestion
    if (nodeTypeForInstanceType(options.instanceType) === NodeType.INFERENTIA) {
```